PL2/PL2-core/src/main/java/nl/tudelft/pl2016gr2/core/algorithms/FilterBubbles.java

```diff
@@ -0,0 +1,149 @@
+package nl.tudelft.pl2016gr2.core.algorithms;
+
+import nl.tudelft.pl2016gr2.model.Bubble;
+import nl.tudelft.pl2016gr2.model.BubbledGraph;
+import nl.tudelft.pl2016gr2.model.GraphInterface;
+import nl.tudelft.pl2016gr2.model.IPhylogeneticTreeNode;
+import nl.tudelft.pl2016gr2.model.Node;
+import nl.tudelft.pl2016gr2.model.OriginalGraph;
+import nl.tudelft.pl2016gr2.test.utility.TestId;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.PriorityQueue;
+import java.util.Queue;
+
+/**
+ * Class used to filter bubbles based on the phylogenetic tree. 
+ * @author Casper
+ *
+ */
+public class FilterBubbles {
+  
+  //private GraphInterface graph;
+  private OriginalGraph originalGraph;
+  private IPhylogeneticTreeNode treeRoot;
+  private HashSet<Integer> allSharedNodes = new HashSet<>();
+  private int mutationId;
+  
+  /**
+   * Creates a FilterBubbles object, with the graph to be filtered and the node of the phylogenetic
+   * tree based on which this graph will be filtered.
+   * @param originalGraph : the graph to be filtered.
+   * @param root : the node of the phylogenetic tree to use for filtering. This will be the root of 
+   *     the tree when a full overview is wanted.
+   */
+  public FilterBubbles(OriginalGraph originalGraph, IPhylogeneticTreeNode root) {
+    //this.graph = graph;
+    this.originalGraph = originalGraph;
+    this.treeRoot = root;
+    mutationId = 10000;
```

Comment:
Is this a hardcoded value for a unique ID, which will break when we load a graph with more than 10 000 nodes? If this is the case you should store a "highestId" value in the OriginalGraph class and create a new unique ID based on that, or maybe use negative IDs for bubbles and positive IDs for nodes (not sure if this will work, as I didn't write this part of the code, but this could be a nice workaround).